docs/bsp.md

```diff
@@ -251,24 +251,22 @@ trait BuildTargetIdentifer {
 
 ### Task Id 
 
-The Task Id allows clients to uniquely identify a resource and establish a client-parent
-relationship with another id.
+The Task Id allows clients to *uniquely* identify a BSP task and establish a client-parent
+relationship with another task id.
 
 ```scala
 trait TaskId {
-  /** The id */
+  /** The id, it has to be unique. */
   def id: String
 
   /** The parent id. */
   def parent: Option[String]
 }
 ```
 
-A task id can represent any child-parent relationship established by the build tool.
-
-An example of use of task ids is logs, where BSP clients can use the task ids of logs
-to improve their readability in the user interface. Clients can show logs in a tree fashion, for
-example, or with dropdowns.
+Clients can use Task Ids to improve the user experience. For example, a client can display
```

Comment:
> to improve the user experience

This can be said about every feature in BSP, I would leave it out.

I propose to remove this paragraph entirely and update the `parent` docstring to say

> The parent task id, if any. A non-empty parent id means this task is a sub-task of the parent task. The child-parent relationship of tasks makes it possible to render tasks in a tree-like user interface.